### PR TITLE
Mb 1563 tab navigation component

### PR DESCRIPTION
**Pull request by gidjin**
## Description

The main task in this story is to add a Tab Navigation component. As this is the first component in our new process this PR also includes some ground work commits. The main one is installing [Loki](https://loki.js.org/) for storybook testing, adding the [Storybook Info Addon](https://www.npmjs.com/package/@storybook/addon-info) to display more details in storybook. I did not add loki tests to circle ci as part of this commit, but I created [MB-1746](https://dp3.atlassian.net/browse/MB-1746) to track that work as well.

## Reviewer Notes

I've added some documentation on running the Loki tests, and started following the new pattern we hope to layout. This is the first of these and I feel like there might be more to refine but wanted to not take too long either.

## Setup

Setup and start storybook
```sh
make client_deps
make storybook
```

Run the loki tests
```sh
make storybook_tests
```

View the TabNav component in browser http://localhost:6006/?path=/story/components--tabnav

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [MB-1563](https://dp3.atlassian.net/browse/MB-1563) for this change
* [Loki](https://loki.js.org/) 

## Screenshots

Screen Shot 2020-02-24 at 12 28 38 PM

### CODE REVIEW

**Comment by samadden** [comment]
The focus state is using Border, which takes up physical space and therefore is pushing elements away after it appears. 

The USWDS spec uses Outline, which doesn't take up space. Can we use this too? Their spec is:

[href]:focus, [tabindex]:focus, iframe:focus {
    outline: .25rem solid  #2491ff;
    outline-offset: 0;
}

Moreover, the Outline treatment in USWDS has equal weight (thickness) on all 4 sides of the box, which is desirable. The current Border treatment has thicker weight on the bottom and right (undesirable).

**Comment by samadden** [comment]
Super persnickety observation:
the blue underline of the Selected tab should sit on top of, not above, the gray border that is the baseline of the entire tab row. In other words, you shouldn't see the gray underline beneath the Selected tab because the thick blue border is taking up that space.

See screenshot from USWDS https://designsystem.digital.gov for what this looks like:

Screen Shot 2020-02-25 at 10 09 54 AM

**Comment by dlee11** [comment]
The yellow tag text should be black (#3D4551), it is currently white.

**Comment by gidjin** [comment]
@dlee11 @tinyels @samadden I think I've updated the component so it should display as expected.

File: src/components/TabNav/index.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+);
+
+TabNav.propTypes = {
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string,
+      notice: PropTypes.string,
+    }),
+  ).isRequired,
+  children: PropTypes.arrayOf(PropTypes.node).isRequired,
```
Comment (tinyels) [inline src/components/TabNav/index.jsx]:
```suggestion
  children: (props, propName, componentName) => {
  // eslint-disable-next-line security/detect-object-injection
  const prop = props[propName];
  let error;

  if (React.Children.count(prop) === 0) {
    error = new Error(`\`${componentName}\` requires Children.`);
  }
  React.Children.forEach(prop, el => {
    if (error) return;
    if (el.type.name !== 'TabPanel') {
      error = new Error(`\`${componentName}\` children must be \`TabPanel\`.`);
    }
  });

  return error;
};
```

Comment (tinyels) [reply src/components/TabNav/index.jsx]: I also wonder if we want to provide some sort of wrapper around `TabPanel` so the dependency on `react-tabs` is isolated to calls here.

Comment (tinyels) [reply src/components/TabNav/index.jsx]: We could also use `childrenOf` from https://github.com/airbnb/prop-types

Comment (gidjin) [reply src/components/TabNav/index.jsx]: That makes a lot of sense let me make those changes and update the pr.

**Pull request merged**